aladdin/config.py

```diff
@@ -39,3 +49,104 @@ def load_config_from_file(file):
+
+    NOTE:
+    If this function returns False it will short-circuit execution of
+    any aladdin command, so returning False should be preceded by some
+    user-friendly error statement about why we're exiting
+    """
+    if os.getenv("ALADDIN_CONFIG_DIR"):
+        # Aladdin config is already set, nothing to do here
+        return True
+
+    err_message = (
+        "Unable to find config repo. "
+        "Please use "
+        "'aladdin config set config_repo <git@github.com:{git_account}/{repo}.git>' "
+        "to set config repo"
+    )
+    try:
+        config = load_user_config_file()
+    except FileNotFoundError:
+        logger.error(err_message)
+        return False
+
+    config_dir: str = config.get("config_dir")
+    config_repo: str = config.get("config_repo")
+    if not config_dir and not config_repo:
+        logger.error(err_message)
+        return False
+
+    if not config_repo:
+        # try to auto-set config_repo
+        if not os.path.isdir(config_dir):
+            logger.error(err_message)
+            return False
+        with working_directory(config_dir):
+            try:
+                remote = subprocess.run(
+                    "git remote get-url origin".split(),
+                    check=True,
+                    capture_output=True,
+                    encoding="utf-8"
+                ).stdout.strip()
+            except subprocess.CalledProcessError:
+                logger.error(err_message)
+                return False
+        *_, git_account, repo = remote.split("/")
+        config_repo = f"git@github.com:{git_account}/{repo}"
+        config["config_repo"] = config_repo
+        set_user_config_file(config)
+
+    remote_config_path = pathlib.Path.home() / ".aladdin/remote_config"
+    commands = [f"git clone -b {__version__} {config_repo} remote_config"]
+    cwd = pathlib.Path.home() / ".aladdin"
+    if os.path.isdir(remote_config_path) and os.path.isdir(remote_config_path / ".git"):
+        cwd = remote_config_path
+        commands = [
+            "git fetch --tags --prune -f",
+            f"git checkout {__version__}"
+        ]
+    with working_directory(cwd):
+        for command in commands:
+            try:
+                subprocess.run(
+                    command.split(),
+                    check=True,
+                    encoding="utf-8",
+                    capture_output=True
+                )
+            except subprocess.CalledProcessError as e:
+                logger.error(
+                    "Failed to fetch aladdin config (%s) from remote: \n%s",
+                    config_repo,
+                    e.stderr.strip() or e.stdout.strip()
+                )
+                return False
+
+    os.environ["ALADDIN_CONFIG_DIR"] = str(remote_config_path)
+
+    if strtobool(os.getenv("ALADDIN_DEV", "false")) and config_dir and os.path.isdir(config_dir):
+        """
+        Allow aladdin developers to use a custom config
```

Comment:
might be helpful to add some notes in the README about this customization via `$ALADDIN_DEV`, now that `--dev` is gone

Reply:
done 👍